### PR TITLE
Implement recombination functions.

### PR DESCRIPTION
**Pull request by michall-m**
Implement some functions for real-domain problems.

- average recombine
- random recombine
- cut point recombine

### CODE REVIEW

**Review by jerzywilczek — APPROVED** [review]
:rocket:

File: src/conf_functions/recombination_functions.rs [hunk 1]
```diff
@@ -1,12 +1,29 @@
...
+    children_genes
+}
+
 pub fn random_cut_point_recombine<const N: usize>(
     parents_genes: (&[f64; N], &[f64; N]),
     rng: &mut R,
 ) -> ([f64; N], [f64; N]) {
     let cut_point = rng.gen_range(0..N);
-    let mut children_genes = ([0.0; N], [0.0; N]);
+    let mut children_genes: ([f64; N], [f64; N]) = ([0.0; N], [0.0; N]);
```
Comment (stlenyk) [inline src/conf_functions/recombination_functions.rs]: Remove type and let the compiler infer it
```suggestion
    let mut children_genes: = ([0.0; N], [0.0; N]);
```

**Review by stlenyk — CHANGES_REQUESTED** [review]
🚀

File: src/conf_functions/recombination_functions.rs (hunk 1 above)
Comment (jerzywilczek) [reply src/conf_functions/recombination_functions.rs]: sry for missing that

**Review by jerzywilczek — APPROVED** [review]
:cowboy_hat_face:

File: src/conf_functions/recombination_functions.rs (hunk 1 above)
Comment (michall-m) [reply src/conf_functions/recombination_functions.rs]: @stlenyk removed

**Review by stlenyk — APPROVED** [review]

**Pull request merged**